src/braket/aws/aws_device.py

```diff
@@ -387,7 +385,7 @@ def topology_graph(self) -> DiGraph:
 
             >>> print(device.topology_graph.edges)
         """
-        return self._topology_graph
+        return self._construct_topology_graph()
```

Comment:
Do we need to cache the results for either of these?

Reply:
The use of caching may conflict with `refresh_metadata`. Benchmarking has not shown a noticeable decrease in performance using non cached data.